### PR TITLE
Fix the #current version to drive off a specific git file instead of …

### PR DESCRIPTION
**Pull request by lmckenzi**
…the file used to manage the list of CI builds - as the former doesn't ensure that only the 'authoritative' project can be the 'current' version for a given IG.  (This is a temporary measure until we update the package server to manage #current.)

### CODE REVIEW

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java [hunk 1]
```diff
@@ -521,26 +521,20 @@ private InputStream fetchFromUrlSpecific(String source, boolean optional) throws
 
   private InputStreamWithSrc loadFromCIBuild(String id) throws IOException {
     checkBuildLoaded();
-    if (ciList.containsKey(id)) {
-      InputStream stream = fetchFromUrlSpecific(Utilities.pathURL(ciList.get(id), "package.tgz"), false);
-      return new InputStreamWithSrc(stream, Utilities.pathURL(ciList.get(id), "package.tgz"), "current");
+    if (packageRepos.containsKey(id)) {
+      InputStream stream = fetchFromUrlSpecific(Utilities.pathURL((String)packageRepos.get(id), "package.tgz"), false);
+      return new InputStreamWithSrc(stream, Utilities.pathURL((String)packageRepos.get(id), "package.tgz"), "current");
```
Comment (markiantorno) [inline org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Do you actually need to cast this to a String? The HashMap is typed already?

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java [hunk 2]
```diff
@@ -637,35 +631,31 @@ private boolean checkBuildLoaded() {
...
+      packageRepos.put(packageId, repo);
     }
-    Collections.sort(builds, new BuildRecordSorter());
-    for (BuildRecord bld : builds) {
+/*    for (BuildRecord bld : builds) {
       if (!ciList.containsKey(bld.getPackageId())) {
         ciList.put(bld.getPackageId(), "https://build.fhir.org/ig/" + bld.getRepo());
       }
-    }
+    }*/
```
Comment (markiantorno) [inline org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Delete the commented out code please.

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java [hunk 3]
```diff
@@ -678,7 +668,7 @@ private Date readDate(String s) {
       e.printStackTrace();
       return new Date();
     }
-  }
+  }*/
```
Comment (markiantorno) [inline org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Delete commented out code.

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java [hunk 4]
```diff
@@ -521,26 +521,20 @@ private InputStream fetchFromUrlSpecific(String source, boolean optional) throws
...
   private String getPackageUrlFromBuildList(String packageId) throws IOException {
     checkBuildLoaded();
-    for (JsonElement n : buildInfo) {
-      JsonObject o = (JsonObject) n;
-      if (packageId.equals(JSONUtil.str(o, "package-id"))) {
-        return JSONUtil.str(o, "url");
-      }
-    }
-    return null;
+    return (String)packageRepos.get(packageId);
```
Comment (markiantorno) [inline org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Again, does this actually require the cast?

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java [hunk 5]
```diff
@@ -612,7 +606,7 @@ private NpmPackage checkCurrency(String id, NpmPackage p) throws IOException {
     checkBuildLoaded();
     // special case: current versions roll over, and we have to check their currency
     try {
-      String url = ciList.get(id);
+      String url = (String)packageRepos.get(id);
```
Comment (markiantorno) [inline org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Casting.

**Review by markiantorno — CHANGES_REQUESTED** [review]

**Review by grahamegrieve — COMMENTED** [review]
I don't like this change at all. I think it's a bad way to solve whatever problem is actually being solved for. Further discussion on Zulip

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java (hunk 1 above)
Comment (lmckenzi) [reply org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Fixed

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java (hunk 4 above)
Comment (lmckenzi) [reply org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Fixed

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java (hunk 5 above)
Comment (lmckenzi) [reply org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Fixed

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java (hunk 2 above)
Comment (lmckenzi) [reply org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Fixed

File: org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java (hunk 3 above)
Comment (lmckenzi) [reply org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/npm/FilesystemPackageCacheManager.java]: Fixed

**Comment by lmckenzi** [comment]
@grahamegrieve I'm ok with a different path, but we need to resolve this soon as, at present, there's a security risk.

**Comment by grahamegrieve** [comment]
This PR isn't going to be accepted. Working with Josh and Mark on an alternative approach

**Pull request closed without merging**